kinesumer.go

```diff
@@ -488,30 +494,37 @@ func (k *Kinesumer) consumePipe(stream string, shard *Shard) {
 				}
 			}
 		case <-checkPointTicker.C:
-			k.commitCheckPoint(stream, shard.ID, lastSequence)
+			k.Commit(stream, shard.ID, lastSequence)
 		}
 	}
 }
 
-// update checkpoint using sequence number.
-func (k *Kinesumer) commitCheckPoint(stream, shardID, lastSeqNum string) {
-	if lastSeqNum == "" {
+// Commit updates check point with sequence number.
+func (k *Kinesumer) Commit(stream, shardID, sequenceNumber string) {
```

Comment:
How about providing the "Mark" or "MarkRecord" function with Commit? Mark will do mark the records as read state (like tiny in-memory cache for checkpoint). Then, when call the Commit function, Commit will update the checkpoint the sequence number of the last marked record. 

In high traffic applications, commit record by record can cause too-many write to state store, so Mark could be useful.